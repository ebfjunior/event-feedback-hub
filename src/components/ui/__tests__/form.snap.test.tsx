import React from 'react';
import { render } from '@testing-library/react';
import { useForm, FormProvider } from 'react-hook-form';
import { FormField, FormItem, FormLabel, FormControl, FormDescription, FormMessage } from '../form';
import { Input } from '../input';

function Demo() {
	const methods = useForm<{ name: string }>({ defaultValues: { name: '' } });
	return (
		<FormProvider {...methods}>
			<form>
				<FormField
					name="name"
					render={() => (
						<FormItem>
							<FormLabel>Name</FormLabel>
							<FormControl>
								<Input />
							</FormControl>
							<FormDescription>desc</FormDescription>
							<FormMessage>error</FormMessage>
						</FormItem>
					)}
				/>
			</form>
		</FormProvider>
	);
}

describe('ui/Form (snapshot)', () => {
	it('renders', () => {
		const { container } = render(<Demo />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


