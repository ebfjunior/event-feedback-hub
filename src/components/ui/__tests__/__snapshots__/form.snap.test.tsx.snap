// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/Form (snapshot) > renders 1`] = `
<form>
  <div
    class="grid gap-2"
    data-slot="form-item"
  >
    <label
      class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
      data-error="false"
      data-slot="form-label"
      for="«r0»-form-item"
    >
      Name
    </label>
    <input
      aria-describedby="«r0»-form-item-description"
      aria-invalid="false"
      class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
      data-slot="form-control"
      id="«r0»-form-item"
    />
    <p
      class="text-muted-foreground text-sm"
      data-slot="form-description"
      id="«r0»-form-item-description"
    >
      desc
    </p>
    <p
      class="text-destructive text-sm"
      data-slot="form-message"
      id="«r0»-form-item-message"
    >
      error
    </p>
  </div>
</form>
`;
